Etg.Yams/Application/AppIdentity.cs

```diff
@@ -18,7 +18,13 @@ public AppIdentity(string id, Version version)
             _version = version;
         }
 
-        public override string ToString()
+		public AppIdentity(string id, string version)
+		{
+			_id = id;
+			_version = Version.Parse(version);
```

Comment:
If `Version.Parse` is not able to parse the version, it'll thrown an Exception:
https://msdn.microsoft.com/en-us/library/system.version.parse(v=vs.110).aspx

I agree that we should more clearly define what version format is supported (at the moment it's whatever Version.Parse can read)